UndertaleModTool/Scripts/Builtin Scripts/RunSwitchAndXboxOnPC.csx

```diff
@@ -0,0 +1,169 @@
+Data.GeneralInfo.Info &= ~(UndertaleGeneralInfo.InfoFlags.Interpolate);
+
+// Use new gamepad functions, because the compatibility ones are completely broken
+
+// use NX routine for joypad detection
+ReplaceTextInGML("gml_Object_obj_time_Create_0", "if (global.osflavor >= 4)", "if (true)");
+ReplaceTextInGML("gml_Object_obj_time_Create_0", """
+    else
+        j_ch = 1
+""", """
+    else if (global.osflavor >= 4)
+        j_ch = 1
+""");
+string os_switch = "os_switch" + (!isXbox ? "_beta" : "");
+ReplaceTextInGML("gml_Object_obj_time_Step_1", "global.osflavor <= 2", "false");
+ReplaceTextInGML("gml_Object_obj_time_Step_1", $"os_type == {os_switch}", "true");
+ReplaceTextInGML("gml_Object_obj_time_Step_1", """
+    if (j_ch > 0)
+        missing_controller_timeout = 0
+""", $"""
+    if (os_type == {os_switch})
+        if (j_ch > 0)
+            missing_controller_timeout = 0
+""");
+
+// Use Xbox default buttons
+ReplaceTextInGML("gml_Object_obj_time_Create_0", """
+global.button0 = 2
+global.button1 = 1
+global.button2 = 4
+""", """
+global.button0 = gp_face1
+global.button1 = gp_face2
+global.button2 = gp_face4
+""");
+ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", """
+    global.button0 = 2
+    global.button1 = 1
+    global.button2 = 4
+""", """
+    global.button0 = gp_face1
+    global.button1 = gp_face2
+    global.button2 = gp_face4
+""");
+
+// axis check
+ReplaceTextInGML("gml_Object_obj_time_Step_1", """
+    if (global.osflavor >= 4)
+    {
+        if (gamepad_button_check
+""", """
+    if (true)
+    {
+        if (gamepad_button_check
+""");
+
+// button check
+ReplaceTextInGML("gml_Script_control_update", "else if (obj_time.j_ch > 0)", "else if (false)");
+ReplaceTextInGML("gml_Script_control_update", "global.osflavor >= 4", "obj_time.j_ch > 0");
+
+// Fix Joystick Menu
+ReplaceTextInGML("gml_Object_obj_joypadmenu_Create_0", "joystick_has_pov(obj_time.j_ch)", "true");
+ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", "joystick_has_pov(obj_time.j_ch)", "true");
+// gamepad_button_count(obj_time.j_ch - 1) might work better but I'm not sure
+ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", "joystick_buttons(obj_time.j_ch)", "11 + 1");
+ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", "joystick_check_button(obj_time.j_ch, i)", "gamepad_button_check(obj_time.j_ch - 1, gp_face1 + i)");
+for (var i = 0; i < 3; i++)
+{
+    ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", $"global.button{i} = i", $"global.button{i} = gp_face1 + i");
+    ReplaceTextInGML("gml_Object_obj_joypadmenu_Draw_0", $"string_hash_to_newline(global.button{i})", $"string(global.button{i} - gp_face1)");
+}
+
+// Use Xbox or PS4 button sprites
+if (isXbox)
+    ReplaceTextInGML("gml_Script_scr_getbuttonsprite", "os_type == os_xboxone", "true");
+else
+    ReplaceTextInGML("gml_Script_scr_getbuttonsprite", "os_type == os_ps4", "true");
+
+// Allow gamepad input for left/right heart halfs
+foreach (var half in new char[]{'l', 'r'})
```

Comment:
I think `in new ['l', 'r']` should work? unsure right now.

Reply:
Or for C#12 it's probably possible to `in ['l', 'r']